nbexchange/plugin/collect.py

```diff
@@ -30,17 +30,31 @@ def download(self, submission, dest_path):
         self.log.debug(
             f"Got back {r.status_code}  {r.headers['content-type']} after file download"
         )
-        tgz = r.content
 
-        try:
-            tar_file = io.BytesIO(tgz)
-            with tarfile.open(fileobj=tar_file) as handle:
-                handle.extractall(path=dest_path)
-        except Exception as e:  # TODO: exception handling
-            if hasattr(e, "message"):
-                self.fail(e.message)
+        if r.status_code > 399:
+            self.fail(
+                f"Error looking for assignments to collect: status code {r.status_code}"
+            )
+
+        if r.headers["content-type"] == "application/x-tar":
+            tgz = r.content
+
+            try:
+                tar_file = io.BytesIO(tgz)
+                with tarfile.open(fileobj=tar_file) as handle:
+                    handle.extractall(path=dest_path)
+            except Exception as e:  # TODO: exception handling
+                if hasattr(e, "message"):
+                    self.fail(e.message)
+                else:
+                    self.fail(e)
+        else:
+            # Fails, even if the json response is a success (for now)
+            data = r.json()
+            if not data["success"]:
+                self.fail("Error looking for assignments to collect")
             else:
-                self.fail(e)
```

Comment:
Same with these fails too. Otherwise things might be pretty hard to debug. If there the other information can be obtained from the logs then don't bother, but it's hard to tell if that's happening.

Reply:
The logs will be split between the exchange, the notebook server-side, and the notebook client-side.
There's no harm in repeating the data :)